diff --git a/drivers/sstar/serial/ms_uart.c b/drivers/sstar/serial/ms_uart.c
index e6ffdbf..3b857af 100644
--- a/drivers/sstar/serial/ms_uart.c
+++ b/drivers/sstar/serial/ms_uart.c
@@ -1936,8 +1936,15 @@ static void ms_uart_set_termios(struct uart_port *p, struct ktermios *pTermios_s
             ms_uart_set_divisor(p, divisor);
         }
     }
+    //#define UART_FCR_TRIGGER_RX_L3 0x1000	   This is unused flag, lets use it to be able to change UART Read Behavior
+    if (pTermios_st->c_iflag & 0x1000){
+        OUTREG8(REG_IIR_FCR(p), UART_FCR_FIFO_ENABLE | UART_FCR_TRIGGER_TX_L1 | UART_FCR_TRIGGER_RX_L3);
+        pr_info("ms_uart_set_termios: set UART_FCR_TRIGGER_RX_L3 : 0x%x\n",pTermios_st->c_iflag);            
+    }else{
+        OUTREG8(REG_IIR_FCR(p), UART_FCR_FIFO_ENABLE | UART_FCR_TRIGGER_TX_L1 | UART_FCR_TRIGGER_RX_L0);
+        pr_info("ms_uart_set_termios: set UART_FCR_TRIGGER_RX_L0 : 0x%x\n",pTermios_st->c_iflag);            
+    }
 
-    OUTREG8(REG_IIR_FCR(p), UART_FCR_FIFO_ENABLE | UART_FCR_TRIGGER_TX_L1 | UART_FCR_TRIGGER_RX_L0);
     INREG8(REG_DLL_THR_RBR(p));
 
     if(p->dev != NULL){
